### PR TITLE
added the new lock manager based on ConcurrentHashMap #8649

### PR DESCRIPTION
**Pull request by taburet**

### CODE REVIEW

File: tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+  public static class ResourceRead implements Callable<Void> {
+    AtomicInteger countRead = new AtomicInteger(0);
+
+    @Override
+    public Void call() throws Exception {
+      lockMgr.acquireSharedLock(this);
+      try {
+        countRead.incrementAndGet();
+        // try {
```
Comment (laa) [inline tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java]: Please remove comment outed code

File: tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+    @Override
+    public Void call() throws Exception {
+      lockMgr.acquireExclusiveLock(this);
+      try {
+        countWrite.incrementAndGet();
+        // try {
+        // Thread.sleep(1 + Math.abs(new Random().nextInt() % 3));
+        // } catch (Exception e) {
+        // }
```
Comment (laa) [inline tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java]: Please remove comment outed code

File: tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+    }
+
+    void write() {
+      lockMgr.acquireExclusiveLock(this);
+      try {
+        countWrite.incrementAndGet();
+        // try {
+        // Thread.sleep(1 + Math.abs(new Random().nextInt() % 3));
+        // } catch (Exception e) {
+        // }
```
Comment (laa) [inline tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java]: Please remove comment outed code

File: tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+    }
+
+    void read() {
+      lockMgr.acquireSharedLock(this);
+      try {
+        countRead.incrementAndGet();
+        // while (countRead < 3) {
+        // // wait for 3 concurrent threads at this point.
+        // Thread.yield();
+        // }
```
Comment (laa) [inline tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java]: Please remove comment outed code

File: tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java (hunk 1 above)
Comment (taburet) [inline tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java]: fixed

File: tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java (hunk 3 above)
Comment (taburet) [inline tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java]: fixed

File: tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java (hunk 4 above)
Comment (taburet) [inline tests/src/test/java/com/orientechnologies/orient/test/database/auto/OIndexOneEntryPerKeyLockManagerTest.java]: fixed

**Pull request merged**